File: umwix1002/umwix1002-fundamental/src/main/java/com/umwix1002/solution/lab/Lab4/L4Q6.java
package Lab4;

public class L4Q6 {
	public static void main(String[] args) {
		int num = (int)(Math.random() * Integer.MAX_VALUE + 1), digit;		
		System.out.println("Random number = " + num);
		for(int i = 0 ; num > 0; num /= 10, i++) {
			digit = num % 10;
			System.out.printf("%15d = %d\n", (1 * (int) Math.pow(10, i)), digit);
		}
	}
}

File: umwix1002-TypeScript/lab/lab04/l4q6.ts
import {createRandom, Random} from "../utils/random";

async function l4q6(): Promise<void> {

    function splitDigitsV1(value: number): string {
        const digits: number[] = [];
        let remainingValue = value;

        while (remainingValue > 0) {
            const lastDigit = remainingValue % 10;

            digits.push(lastDigit);
            remainingValue = Math.floor(remainingValue / 10);
        }

        return digits.reverse().join(",");
    }

    function splitDigitsV2(value: number): string {
        return String(value)
            .split("")
            .join(",")
    }

    async function main(): Promise<void> {
        const random: Random = createRandom();

        const randomValue: number = random.nextInteger(0);

        console.log(`Random Value: ${randomValue}, Digits: ${splitDigitsV1(randomValue)}`);
        console.log(`Random Value: ${randomValue}, Digits: ${splitDigitsV2(randomValue)}`);
    }

    await main();
}

l4q6()
    .then(() => process.exit(0))
    .catch((error: unknown) => {
        console.error(error);
        process.exit(1);
    });